packages/statusbar/src/browser/statusbar-service.ts

```diff
@@ -0,0 +1,58 @@
+/*
+* Copyright (C) 2017 TypeFox and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+import { injectable, inject } from "inversify";
+import { StatusbarWidget } from './statusbar-widget';
+import { Command, Disposable, DisposableCollection } from "@theia/core";
+import { WidgetManager } from "@theia/core/lib/browser";
+
+export interface StatusbarEntry {
+    text: string;
+    tooltip?: string;
+    command?: Command;
+    arguments?: any[];
+}
+
+export enum StatusbarAlignment {
+    LEFT, RIGHT
+}
+
+export const STATUSBAR_WIDGET_FACTORY_ID = 'statusbar';
+
+@injectable()
+export class StatusbarService {
+
+    protected toDispose = new DisposableCollection();
+
+    constructor( @inject(WidgetManager) protected readonly widgetManager: WidgetManager) {
+
+    }
+
+    async getOrCreateWidget(id: string): Promise<StatusbarWidget> {
```

Comment:
we should not expose the actual widget, but instead use the id always.
We also only need to methods:
 - `setElement(id:string, data: StatusBarData, alignment: StatusbarAlignment, priority?: number)`
 - `removeElement(id: string)`

Reply:
Also, we discussed, to have the statusbar as part of the shell, just like the sidebars and main area, as we will need to store the inner state between sessions.
A user would use the shell directly to change the status bar, e.g.:
`shell.statusbar.setEntry('errors', errorEntry, StatusbarAlignment.LEFT, 2)`